elastic/assets/configuration/spec.yaml

```diff
@@ -110,6 +110,35 @@ files:
         display_default: false
         example: true
       enabled: true
+    - name: custom_queries
+      description: |
+        Run custom queries for Elasticsearch. Each custom query endpoint can collect multiple metrics and tags.
+        Note: Each custom query requires an `endpoint` and a `metrics` parameters, and each `metrics` parameter should
+        have at least one metric containing `datadog_metric_name` and `es_metric_name`. `type` is an optional parameters
+        and defaults to `gauge`.
```

Comment:
1/ The `es_metric_name` would be the path to the actual value. In the case of `elasticsearch.devices_count`, the `es_metric_name` would be `aggregations.genres.buckets.doc_count`, but I can see that I need to account for possibly multiple metrics with that name. As for the `tag`, I can add the ability to use an `es_metric_name` as a tag, as in setting `aggregations.genres.buckets.key` as a tag for `org_id` in your example.

2/ I was looking into QueryManager as a possible way of implementing custom queries, although I wasn't sure if it was viable. The current ES check uses HTTP requests to fetch metrics, so adding QueryManager could mean either only using QueryManager for custom queries or rewriting the existing check to use QueryManager entirely. WDYT?

Reply:
1/ agree with your points.

2/ It seems the QueryManager expected us to pass a function to be executed, e.g. execute_query_raw here : https://github.com/DataDog/integrations-core/blob/634d567f3c38d32aabb3f4c16b50bcfa8a4ae0fb/snowflake/datadog_checks/snowflake/check.py#L72

But there is possible a need to match the expected row format of the QueryManager

Reply:
Expanding QueryManager for NoSQL database checks would be too complicated, since many NoSQL databases function differently, so the logic would have to be adapted for each implementation, which would eliminate the benefit of using QueryManager. After discussing with the team, I believe the best option is to implement custom logic for Elasticsearch.